modules/light-clients/08-wasm/types/migrate_contract_test.go

```diff
@@ -0,0 +1,140 @@
+package types_test
+
+import (
+	"crypto/sha256"
+	"encoding/hex"
+	"encoding/json"
+
+	wasmvm "github.com/CosmWasm/wasmvm"
+	wasmvmtypes "github.com/CosmWasm/wasmvm/types"
+
+	"github.com/cosmos/ibc-go/modules/light-clients/08-wasm/internal/ibcwasm"
+	wasmtesting "github.com/cosmos/ibc-go/modules/light-clients/08-wasm/testing"
+	"github.com/cosmos/ibc-go/modules/light-clients/08-wasm/types"
+	clienttypes "github.com/cosmos/ibc-go/v8/modules/core/02-client/types"
+	host "github.com/cosmos/ibc-go/v8/modules/core/24-host"
+)
+
+func (suite *TypesTestSuite) TestMigrateContract() {
+	var (
+		oldHash        [32]byte
+		newHash        [32]byte
+		payload        []byte
+		expClientState *types.ClientState
+	)
+
+	testCases := []struct {
+		name     string
+		malleate func()
+		expErr   error
+	}{
+		{
+			"success: new and old code hash are different",
+			func() {
+				newHash = sha256.Sum256([]byte{1, 2, 3})
+				err := ibcwasm.CodeHashes.Set(suite.chainA.GetContext(), newHash[:])
+				suite.Require().NoError(err)
+
+				payload = []byte{1}
+				expChecksum := wasmvmtypes.ForceNewChecksum(hex.EncodeToString(oldHash[:]))
+
+				suite.mockVM.MigrateFn = func(codeID wasmvm.Checksum, env wasmvmtypes.Env, msg []byte, _ wasmvm.KVStore, _ wasmvm.GoAPI, _ wasmvm.Querier, _ wasmvm.GasMeter, _ uint64, _ wasmvmtypes.UFraction) (*wasmvmtypes.Response, uint64, error) {
+					suite.Require().Equal(expChecksum, codeID)
+					suite.Require().Equal(defaultWasmClientID, env.Contract.Address)
+					suite.Require().Equal(payload, msg)
+
+					data, err := json.Marshal(types.EmptyResult{})
+					suite.Require().NoError(err)
+
+					return &wasmvmtypes.Response{Data: data}, wasmtesting.DefaultGasUsed, nil
+				}
+			},
+			nil,
+		},
+		{
+			"success: update client state",
+			func() {
+				newHash = sha256.Sum256([]byte{1, 2, 3})
+				err := ibcwasm.CodeHashes.Set(suite.chainA.GetContext(), newHash[:])
+				suite.Require().NoError(err)
+
+				suite.mockVM.MigrateFn = func(_ wasmvm.Checksum, _ wasmvmtypes.Env, _ []byte, store wasmvm.KVStore, _ wasmvm.GoAPI, _ wasmvm.Querier, _ wasmvm.GasMeter, _ uint64, _ wasmvmtypes.UFraction) (*wasmvmtypes.Response, uint64, error) {
+					expClientState = types.NewClientState([]byte{1}, newHash[:], clienttypes.NewHeight(2000, 2))
+					store.Set(host.ClientStateKey(), clienttypes.MustMarshalClientState(suite.chainA.App.AppCodec(), expClientState))
```

Comment:
should this have the same assertions as the success case?

Reply:
I think this is what we did in other tests too. The point of sucess case is to assert right params were passed but I don't think we need to check env, migrateMsg and etc again in here